src/main/java/org/jabref/websocket/JabRefWebsocketServer.java

```diff
@@ -0,0 +1,476 @@
+package org.jabref.websocket;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+
+import org.jabref.websocket.handlers.HandlerCmdRegister;
+import org.jabref.websocket.handlers.HandlerInfoGoogleScholarCitationCounts;
+import org.jabref.websocket.handlers.HandlerInfoMessage;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import org.java_websocket.WebSocket;
+import org.java_websocket.handshake.ClientHandshake;
+import org.java_websocket.server.WebSocketServer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A simple, robust websocket server implementation for JabRef for bidirectional communication with arbitrarily many
+ * websocket clients
+ */
+public class JabRefWebsocketServer extends WebSocketServer {
+    // internals
+    private static final int MAX_ONMESSAGE_CALLS_IN_PARALLEL = 500; // default: 500; 1: enables sequential processing
+    private static final int DEFAULT_PORT = 8855;
+    private static final Logger LOGGER = LoggerFactory.getLogger(JabRefWebsocketServer.class);
+    private static final boolean SHOW_VERBOSE_DEBUG_OUTPUT = false;
+
+    private static JabRefWebsocketServer jabRefWebsocketServerSingleton = null;
```

Comment:
We are trying to get rid off singleton instances since they make testing very hard. Instead please create one instance in `Globals` and then pass it along (similarly to how it is done with the DialogService).